eng/common/cross/toolchain.cmake

```diff
@@ -48,11 +48,13 @@ elseif(TARGET_ARCH_NAME STREQUAL "arm64")
   set(CMAKE_SYSTEM_PROCESSOR aarch64)
   if(EXISTS ${CROSS_ROOTFS}/usr/lib/gcc/aarch64-alpine-linux-musl)
     set(TOOLCHAIN "aarch64-alpine-linux-musl")
-  else()
+  elseif(LINUX)
```

Comment:
I believe this one is taken from https://github.com/dotnet/arcade/pull/9834

Reply:
@janvorli, in this context, we need to detect Alpine Linux, which is why it is detected before LINUX. `eng/common` changes are mirror of https://github.com/dotnet/arcade/pull/9834.

Reply:
I know, I had this structure on my mind:
```cmake
if(LINUX)
    if(EXISTS ${CROSS_ROOTFS}/usr/lib/gcc/aarch64-alpine-linux-musl)
        set(TOOLCHAIN "aarch64-alpine-linux-musl")
    else()
        set(TOOLCHAIN "aarch64-linux-gnu")
    endif()
elseif(FREEBSD)
...
endif()
```

Reply:
Stupid question - won't some of the changes by overwritten by arcade in some point in time - I don't know the flow of those, but I see some files are the same and looking at history they are updated?

Reply:
It will be overwritten. I don't have problem with adhering to proposed strucure, but I think @janvorli is fine with the current structure, so lets keep it as is for now (they are logically doing the same thing).

We can restructure toolchain.cmake separately and address other feedback (including bringing FreeBSD on same `set(TOOLCHAIN ...` plan as other platforms etc.)